api/workload/v2/workload.go

```diff
@@ -0,0 +1,219 @@
+package workload
+
+import (
+	"context"
+	"crypto"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/spiffe/spire/proto/spire/api/workload"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	// DefaultAgentAddress is the default GRPC address to contact the spire agent at.
+	DefaultAgentAddress = "unix:///tmp/agent.sock"
+
+	_unixPathPrefix = "unix://"
+)
+
+// X509SVIDs is an X.509 SVID response from the SPIFFE Workload API.
+type X509SVIDs struct {
+	// SVIDs is a list of X509SVID messages, each of which includes a single
+	// SPIFFE Verifiable Identity Document, along with its private key and bundle.
+	SVIDs []*X509SVID
+
+	// CRL is a list of revoked certificates.
+	// Unimplemented.
+	CRL *pkix.CertificateList
+
+	// FederatedBundles are CA certificate bundles belonging to foreign Trust Domains
+	// that the workload should trust, keyed by the SPIFFE ID of the foreign domain.
+	// Unimplemented.
+	FederatedBundles map[string]*x509.CertPool
+}
+
+// Default returns the default SVID (the first in the list).
+//
+// See the SPIFFE Workload API standard Section 5.3
+// (https://github.com/spiffe/spiffe/blob/master/standards/SPIFFE_Workload_API.md#53-default-identity)
+func (x *X509SVIDs) Default() *X509SVID {
+	return x.SVIDs[0]
+}
+
+// SVID is an X.509 SPIFFE Verifiable Identity Document.
+//
+// See https://github.com/spiffe/spiffe/blob/master/standards/X509-SVID.md
+type X509SVID struct {
+	SPIFFEID     string
+	PrivateKey   crypto.Signer
+	Certificates []*x509.Certificate
+	TrustBundle  *x509.CertPool
```

Comment:
CertPool is convenient for verify opts, but highly inconvenient for accessing the certs themselves (unexported field). Better to have the trust bundle field be a cert array, though could have convenience method that gives you a cert pool if desired.

Reply:
I've changed `TrustBundle` to be a list of certs, with a `TrustBundePool() *x509.CertPool` convenience method